theatre/studio/src/panels/DetailPanel/prop-editors/DetailDeterminePropEditor.tsx

```diff
@@ -0,0 +1,60 @@
+import type {
+  PropTypeConfig,
+  PropTypeConfig_AllSimples,
+} from '@theatre/core/propTypes'
+import type SheetObject from '@theatre/core/sheetObjects/SheetObject'
+import type {Pointer} from '@theatre/dataverse'
+import React from 'react'
+import DetailCompoundPropEditor from './DetailDeterminePropEditor/DetailCompoundPropEditor'
+import DetailSimplePropEditor from './DetailDeterminePropEditor/DetailSimplePropEditor'
+import {getPropTypeByPointer} from '@theatre/studio/panels/DetailPanel/prop-editors/utils/getPropTypeByPointer'
+import {simplePropEditorByPropType} from '@theatre/studio/panels/DetailPanel/prop-editors/simple-editors/simplePropEditorByPropType'
+import type {PropConfigForType} from './utils/PropConfigForType'
+import type {ISimplePropEditorReactProps} from './simple-editors/ISimplePropEditorReactProps'
+
+const DetailDeterminePropEditor: React.VFC<
+  IDetailDeterminePropEditorProps<PropTypeConfig['type']>
+> = (p) => {
+  const propConfig =
+    p.propConfig ?? getPropTypeByPointer(p.pointerToProp, p.obj)
+
+  if (propConfig.type === 'compound') {
+    return (
+      <DetailCompoundPropEditor
+        obj={p.obj}
+        visualIndentation={p.visualIndentation}
+        pointerToProp={p.pointerToProp}
+        propConfig={propConfig}
+      />
+    )
+  } else if (propConfig.type === 'enum') {
+    // hmm: document how enum gets put into ui (through label?)
```

Comment:
no support for enums yet at any level

Reply:
Okay!

Reply:
Applied comment that "enums not implemented"